scripts/ci/test_help_doc_arguments.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import re
+import sys
+import os
+
+logger = logging.getLogger('az-publish')
+
+
+DEFAULT_TEST_CMDS = []
+
+pattern = r".*?\s+((--\S+)+( --\S+)*( -\S)*).*?:"   # meaning: escape ansi color codes then
+                                        #       match one or more long options ('--') followed by an (optional) short option ('-')
+                                        #       and then the first semicolon after 0 or more chars (':')
+                                        #       Also handles cases where there are multiple long options see --edition in az sql mi
+COMPILED_RE = re.compile(pattern)
+GLOBAL_ARGS = ["debug", "help", "output", "query", "subscription", "verbose"]
+
+
+def main(xml_file, commands=[]):
+
+    if commands is None:
+        commands = ["vm create"]
+    with open(xml_file, 'r') as f:
+        doc_contents = xmltodict.parse(f.read(), process_namespaces=True)
+
+    entries = doc_contents['document']['desc']
+    cmd_to_content_dict = {entry['desc_signature']['desc_addname']['#text']: entry['desc_content'] for entry in
+                           entries if entry['@objtype'] == "clicommand"}
+    for cmd in commands:
+        logger.warning("Testing %s help...", cmd)
+        help_args = get_args_from_help_text(cmd)
+        doc_args = get_args_from_doc_dict(cmd, cmd_to_content_dict)
+        try:
+            assert help_args == doc_args
+        except AssertionError as e:
+            logger.warning("CLI help args: %s\n\nDoc help args: %s\n.", ", ".join(help_args), ", ".join(doc_args))
+            logger.warning(e)
+            exit(1)
+
+
+    logger.info("Doc help arguments match CLI help arguments.")
+
+
+
+def get_args_from_help_text(cmd_name):
+    cmd = "az {} -h".format(cmd_name)
+    cmd_args = cmd.split()
+    completedprocess = subprocess.run(cmd_args, stdout=subprocess.PIPE, encoding="utf-8")
+
+    if completedprocess.returncode != 0:
+        logger.warning("Error running %s", " ".join(cmd))
+        exit(1)
+
+    lines = completedprocess.stdout.splitlines()
+    arguments = set()
+    for line in lines:
+        match = COMPILED_RE.search(line)
+        if match:
+            options = match.group(1)
+            if _get_long_option(options) not in GLOBAL_ARGS:
+                arguments.add(options)
+    return arguments
+
+
+def get_args_from_doc_dict(cmd_name, cmd_to_content_dict):
+    try:
+        arg_info_dict = {possible_arg['desc_signature']['desc_addname']['#text']: possible_arg['desc_content']['field_list']['field']
+                for possible_arg in cmd_to_content_dict[cmd_name]['desc'] if possible_arg['@objtype'] == "cliarg" and
+                _get_long_option(possible_arg['desc_signature']['desc_addname']['#text']) not in GLOBAL_ARGS}
+    except KeyError:
+        return set()
+
+    arguments = set()
+    # Only add arguments that don't have the deprecated field. #todo: tidy up logic.
+    for arg, arg_fields in arg_info_dict.items():
+        if "enable-rbac" in arg:
```

Comment:
What is this?

Reply:
I accidentally left this in from debugging deprecated commands in the doc xml output. I will remove this.